src/miner.erl

```diff
@@ -410,8 +410,20 @@ handle_call({create_block, Stamps, Txns, HBBFTRound}, _From, State) ->
                 LastBlockTimestamp = blockchain_block:time(CurrentBlock),
                 BlockTime = miner_util:median([ X || {_, {X, _}} <- Stamps,
                                                      X > LastBlockTimestamp]),
-                {ValidTransactions, InvalidTransactions} =
-                    blockchain_txn:validate(SortedTransactions, Chain),
+
+                {BundledValidTransactions, InvalidTransactions} = blockchain_txn:validate(SortedTransactions, Chain),
+
+                ValidTransactions = lists:reverse(lists:foldl(fun(Txn, Acc) ->
+                                                                      case blockchain_txn:type(Txn) of
```

Comment:
hm.  I kind of want this code not to live in miner.  can we add it to blockchain_txn?  maybe `blockchain_txn:flatten` or something?

Reply:
we can call it here, I just don't want anything in miner to have to understand those sorts of details.